xla/service/gpu/execution_stream_assignment.cc

```diff
@@ -78,8 +91,24 @@ ExecutionStreamAssignment::ExecutionStreamAssignment(
     // instructions. Asynchronous instructions will be handled afterwards.
     for (HloInstruction* instruction : pending.node->instructions()) {
       if (instruction->IsAsynchronous()) continue;
-      CHECK(sync_instructions_.try_emplace(instruction, pending.stream_id)
-                .second);
+      // The copy-start operation can only be in the main stream (stream 0).
+      if (instruction->opcode() == HloOpcode::kCopyStart) {
+        ExecutionStreamId destination_stream_id = pending.stream_id;
+        CHECK_EQ(pending.stream_id, ExecutionStreamId(0));
+        if (copy_start_stream_id == ExecutionStreamId(0)) {
```

Comment:
It should not be necessary to assign all copy-starts to the same stream, how about getting rid of `copy_start_stream_id`?

Reply:
Done.